### PR TITLE
Add docs for expressions, minor cleanups

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Wrote a first draft about sqlglot's expressions & did some minor fixes here and there, mostly related to the README.

The idea is to create an index for helper expression methods in a later PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -272,7 +272,7 @@ transformed_tree.sql()
 
 ### SQL Optimizer
 
-SQLGlot can rewrite queries into an "optimized" form. It performs a variety of [techniques](sqlglot/optimizer/optimizer.py) to create a new canonical AST. This AST can be used to standardize queries or provide the foundations for implementing an actual engine. For example:
+SQLGlot can rewrite queries into an "optimized" form. It performs a variety of [techniques](https://github.com/tobymao/sqlglot/blob/main/sqlglot/optimizer/optimizer.py) to create a new canonical AST. This AST can be used to standardize queries or provide the foundations for implementing an actual engine. For example:
```
Comment (tobymao) [inline README.md:275]: why is this changed?

Comment (GeorgeSittas) [reply README.md:275]: Because the links are broken when we do `make docs-serve`, it can't navigate the repo.

**Review by tobymao — APPROVED** [review]
